frontend/packages/console-dynamic-plugin-sdk/src/extensions/pages.ts

```diff
@@ -14,7 +33,65 @@ export type StandaloneRoutePage = ExtensionDeclaration<
   }
 >;
 
+/** Adds new page to Console router. */
+export type RoutePage = ExtensionDeclaration<
+  'console.page/route',
+  {
+    /** The React component to load for the page. */
+    component: CodeRef<React.Component<RouteComponentProps>>;
+    /** The perspective ID to which this item belongs to. If not specified, contributes to all perspectives. */
+    perspective?: string;
+    /** Valid URL path or array of paths that `path-to-regexp@^1.7.0` understands. */
+    path: string | string[];
+    /** When true, will only match if the path matches the `location.pathname` exactly. */
+    exact?: boolean;
+    /** When true, a path that has a trailing slash will only match a location.pathname with a trailing slash. This has no effect when there are additional URL segments in the location.pathname. */
+    strict?: boolean;
+    /** When true, will match if the path is case sensitive. */
+    sensitive?: boolean;
+  }
+>;
+
+/** Adds new resource list to Console router. */
+export type ResourceListPage = ExtensionDeclaration<'console.page/resource/list', ResourcePage>;
+
+/** Adds new details page to Console router. */
+export type ResourceDetailsPage = ExtensionDeclaration<
+  'console.page/resource/details',
+  ResourcePage
+>;
+
+/** Adds new resource tab page to Console router. */
+export type ResourceTabPage = ExtensionDeclaration<
+  'console.page/resource/tab',
+  ResourcePage & {
+    /** The name of the tab. */
+    name: string;
+    /** The optional href for the tab link. If not provided, the first `path` is used. */
+    href?: string;
+    /** Valid URL path or array of paths that `path-to-regexp@^1.7.0` understands. */
+    path: string | string[];
+    /** When true, will only match if the path matches the `location.pathname` exactly. */
+    exact?: boolean;
+    /** When true, a path that has a trailing slash will only match a location.pathname with a trailing slash. This has no effect when there are additional URL segments in the location.pathname. */
+    strict?: boolean;
```

Comment:
Please go over all usages of the current `Page/Resource/Tab` extension and see if we really need those additional properties, like `strict` and `sensitive`.

Reply:
could be useful for later usages?

Reply:
We can always add more properties later, I'd recommend to keep the API surface minimal :smiley: